### PR TITLE
test(P4): add coverage for src/stores/ and storage layer (#745)

### LINKED ISSUE #745: test(P4): src/stores/ とストレージ層のテスト追加
## 親 Epic
- #741

## 目的 / Purpose

`src/stores/` は 4 ファイル中 1 ファイル（`mcpConfigStore`）のみテスト済。グローバル状態とストレージ抽象化はデータ整合性の要なので、coverage と回帰防止の双方の観点でテストを追加する。

## 対象ファイル

| ファイル | 必要なテスト観点 |
|---|---|
| `src/stores/pageStore.ts` | ゲストモードの page CRUD、links / ghost links 計算、永続化 |
| `src/stores/aiChatStore.ts` | モデル選択、コンテキストトグル、永続化、reset 動作 |
| `src/lib/storage/index.ts` | プロバイダ factory の分岐、レガシー migration ロジック |
| `src/lib/pageRepository.ts` | リポジトリインターフェース全 method の挙動契約 |

## 作業ステップ / Steps

1. Zustand ストアは `vi.useFakeTimers` と `act()` を組み合わせ、状態更新と persist の両方を検証。
2. `mcpConfigStore.test.ts` の作りに揃え、各 action ごとに「初期状態 → action 呼び出し → 期待状態」を表駆動で確認。
3. `storage/index.ts` の factory は環境変数・引数の組み合わせ表で全分岐を網羅。
4. `pageRepository.ts` はインターフェース層なので、各 method が依存（adapter）に正しく委譲しているかを spy で確認。

## 完了条件 / Done

- [ ] 対象 4 ファイルそれぞれに `*.test.ts` が追加されている。
- [ ] `bun run test:run` がパスする。
- [ ] `pageStore` / `aiChatStore` の全 action が少なくとも 1 ケースずつカバーされている。
- [ ] `storage/index.ts` のローカル mutation score **70% 以上**。

## 参考

- 既存: `src/stores/mcpConfigStore.test.ts`
- ストレージアダプタ実装: `src/lib/storageAdapter/IndexedDBStorageAdapter.test.ts`

### PR DESCRIPTION
**Pull request by otomatty**
- pageStore: ゲスト CRUD / リンク / ゴーストリンク / 永続化 + v1→v2→v3 マイグレーション
- aiChatStore: 全 action と partialize された永続キーの契約
- storage/index: factory 全分岐 + cloudflare-r2 → s3 レガシー migration / アップロード設定
- pageRepository: IPageRepository 契約テスト + onMutate コールバック

Closes #745.

---

<a href="https://app.devin.ai/review/otomatty/zedi/pull/753" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test suites covering page repository behavior (CRUD, soft-delete, listing, search, links/ghost-links and promotion), storage provider selection and upload settings, AI chat state persistence and UI toggles, and page store actions/persistence including migration scenarios and link/ghost-link edge cases to improve stability and prevent regressions.

### CODE REVIEW

File: src/lib/pageRepository.test.ts [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+      return ids.map((id) => store.pages.get(id)).filter((p): p is Page => Boolean(p));
+    },
+    async getPageByTitle(_userId, title) {
+      return [...store.pages.values()].find((p) => p.title === title) ?? null;
+    },
+    async checkDuplicateTitle(_userId, title, excludePageId) {
+      return (
+        [...store.pages.values()].find((p) => p.title === title && p.id !== excludePageId) ?? null
+      );
+    },
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/lib/pageRepository.test.ts:104]: 対応しました（commit `3910aeb`）。

`getPagesSummary` / `getPagesByIds` / `getPageByTitle` / `checkDuplicateTitle` に `isDeleted` フィルタを追加し、read 系全体で soft-delete 挙動が一貫するようにしました。さらに `read methods consistently hide soft-deleted pages` というリグレッションガードのテストも追加しています（`src/lib/pageRepository.test.ts`）。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/lib/pageRepository.test.ts [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+    async deletePage(_userId, pageId) {
+      const p = store.pages.get(pageId);
+      if (p) store.pages.set(pageId, { ...p, isDeleted: true });
+      await fireMutate();
+    },
+    async searchPages(_userId, query) {
+      return [...store.pages.values()].filter((p) =>
+        p.title.toLowerCase().includes(query.toLowerCase()),
+      );
+    },
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/lib/pageRepository.test.ts:125]: 対応しました（commit `3910aeb`）。

モックの `searchPages` を `pageStore.searchPages` の契約に揃えました:

- `title` に加え `content` も部分一致対象に追加
- 空クエリ（`""` / 空白のみ）は `[]` を返却
- `isDeleted` のページは除外

リグレッション用に `searchPages mirrors pageStore semantics: title+content, trim, hide deleted` テストも追加しています。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/lib/pageRepository.test.ts [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+      for (const sourceId of sources) {
+        await repo.addLink(sourceId, created.id, "wiki");
+        await repo.removeGhostLink(linkText, sourceId, "wiki");
+      }
+      return created;
+    },
+  };
+
+  return repo;
+}
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/lib/pageRepository.test.ts:206]: 対応しました（commit `f1d27b6`）。

実プロダクション `StorageAdapterPageRepository` を直接アンカーした薄い契約スイート `StorageAdapterPageRepository (production) satisfies IPageRepository` を追加し、`createPage` / `getPage` / `addLink` / `deletePage` の adapter / API への委譲を spy で検証しています。これで、本ファイルでも実装側のリグレッションを検出できます。

※ in-memory モック `createInMemoryRepository` は **interface の生きたドキュメント**として残しています（`pageRepository.ts` は型のみのファイルで、将来の他実装（例: Tauri 側）が契約に従うかを self-document する目的）。網羅的な adapter 委譲検証は引き続き `src/lib/pageRepository/StorageAdapterPageRepository.test.ts` 側にあり、本ファイルはその上の interface contract layer + production smoke という二層構造になります。

加えて、レビューサマリの 2 件の Nitpick も同コミットで対応しました:

- `onMutate` テストを link / ghost link mutation までカバーし、`expect(onMutate).toHaveBeenCalledTimes(7)` まで拡張。
- `aiChatStore` の rehydrate テストに揮発フィールド (`activeConversationId` / `isStreaming` / `showConversationList`) の assertion を追加し、テスト名と一致させました。

---
_Generated by [Claude Code](https://claude.ai/code)_
